plugin/pkg/admission/imagepolicy/admission.go

```diff
@@ -0,0 +1,255 @@
+		if err != nil {
+			return nil, err
+		}
+		return newImagePolicyWebhook, nil
+	})
+}
+
+// imagePolicyWebhook is an implementation of admission.Interface.
+type imagePolicyWebhook struct {
+	*admission.Handler
+	*webhook.GenericWebhook
+	responseCache *cache.LRUExpireCache
+	allowTTL      time.Duration
+	denyTTL       time.Duration
+	defaultAllow  bool
+}
+
+// imagePolicyWebhookConfig holds config data for imagePolicyWebhook
+type imagePolicyWebhookConfig struct {
+	KubeConfigFile string        `json:"kubeConfigFile"`
+	AllowTTL       time.Duration `json:"allowTTL"`
+	DenyTTL        time.Duration `json:"denyTTL"`
+	DefaultAllow   bool          `json:"defaultAllow"`
+}
+
+// AdmissionConfig holds config data for admission controllers
+type AdmissionConfig struct {
+	Webhook imagePolicyWebhookConfig `json:"webhook"`
+}
+
+func (a *imagePolicyWebhook) statusTTL(status v1beta1.ImageReviewStatus) time.Duration {
+	if status.Allowed {
+		return a.allowTTL
+	}
+	return a.denyTTL
+}
+
+// Filter out annotations that don't match *.image-policy.k8s.io/*
+func (a *imagePolicyWebhook) filterAnnotations(allAnnotations map[string]string) map[string]string {
+	annotations := make(map[string]string)
+	for k, v := range allAnnotations {
+		if strings.Contains(k, ".image-policy.k8s.io/") {
+			annotations[k] = v
+		}
+	}
+	return annotations
+}
+
+// Function to call on webhook failure; behavior determined by defaultAllow flag
+func (a *imagePolicyWebhook) webhookError(err error) error {
+	if err != nil {
+		glog.V(4).Infof("error contacting webhook backend: %s")
+		if a.defaultAllow {
+			glog.V(4).Infof("pod allowed in spite of webhook backend failure")
+			return nil
+		}
+		glog.V(4).Infof("pod not allowed due to webhook backend failure ")
+		return err
+	}
+	return nil
+}
+
+func (a *imagePolicyWebhook) Admit(attributes admission.Attributes) (err error) {
+	// Ignore all calls to subresources or resources other than pods.
+	if len(attributes.GetSubresource()) != 0 || attributes.GetResource().GroupResource() != api.Resource("pods") {
+		return nil
+	}
+	pod, ok := attributes.GetObject().(*api.Pod)
+	if !ok {
+		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
+	}
+
+	annotations := a.filterAnnotations(pod.Annotations)
+	namespace := attributes.GetNamespace()
+
+	// Build list of ImageReviews to send to backend, one per image
+	var imageReviews []v1beta1.ImageReview
+	for _, c := range pod.Spec.Containers {
+		imageReviews = append(imageReviews, v1beta1.ImageReview{
+			Spec: v1beta1.ImageReviewSpec{
+				Container: v1beta1.ImageReviewContainerSpec{
+					Image: c.Image,
+				},
+				Annotations: annotations,
+				Namespace:   namespace,
+			},
+		})
+	}
+
+	for _, r := range imageReviews {
+		if err := a.admitContainer(&r); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (a *imagePolicyWebhook) admitContainer(review *v1beta1.ImageReview) error {
+	cacheKey, err := json.Marshal(review.Spec)
```

Comment:
This should use our scheme/codec encoding path.  This structure here limits our ability to serialize to protobuf as a for instance.